crates/context/config/src/client.rs

```diff
@@ -19,44 +20,39 @@ pub mod near;
 pub mod relayer;
 pub mod starknet;
 
-use config::{ContextConfigClientConfig, ContextConfigClientSelectedSigner, CryptoCredentials};
+use config::{
+    ContextConfigClientConfig, ContextConfigClientSelectedSigner, CryptoCredentials, Protocol,
+};
 
-#[non_exhaustive]
 #[derive(Clone, Debug)]
-pub enum TransportChoice<L, R, T> {
-    Left(L),
-    Right(R),
-    Third(T),
-}
-
 #[non_exhaustive]
-#[derive(Debug)]
-pub enum TransportError<L, R, T> {
+pub enum Either<L, R> {
     Left(L),
     Right(R),
-    Third(T),
 }
 
-impl<L, R, T> std::fmt::Display for TransportError<L, R, T>
+pub type TransportErr<L, R, T> = Either<L, Either<R, T>>;
+
+impl<L, R, T> Display for TransportErr<L, R, T>
 where
-    L: std::fmt::Display,
-    R: std::fmt::Display,
-    T: std::fmt::Display,
+    L: Display,
+    R: Display,
+    T: Display,
 {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
         match self {
-            TransportError::Left(err) => write!(f, "{}", err),
-            TransportError::Right(err) => write!(f, "{}", err),
-            TransportError::Third(err) => write!(f, "{}", err),
+            Self::Left(err) => write!(f, "{err}"),
+            Self::Right(Either::Left(err)) => write!(f, "{err}"),
+            Self::Right(Either::Right(err)) => write!(f, "{err}"),
         }
     }
 }
 
-impl<L, R, T> std::error::Error for TransportError<L, R, T>
+impl<L, R, T> CoreError for TransportErr<L, R, T>
 where
-    L: std::error::Error,
-    R: std::error::Error,
-    T: std::error::Error,
+    L: CoreError,
+    R: CoreError,
+    T: CoreError,
 {
 }
```

Comment:
`Either` [impls](https://docs.rs/either/latest/either/enum.Either.html#impl-Display-for-Either%3CL,+R%3E) `Display`, let's reintroduce the crate

it'll save us from manually writing this

Reply:
Yeah you are right about this, missed import along the way. Updated and code cleanup